analyzer_test.go

```diff
@@ -49,17 +49,76 @@ func TestDeriveManifestAndLock(t *testing.T) {
 	}
 }
 
-func TestDeriveManifestAndLockDoesNotExist(t *testing.T) {
-	dir, err := ioutil.TempDir("", "dep")
+func TestAnalyzerDeriveManifestAndLockDoesNotExist(t *testing.T) {
+	h := test.NewHelper(t)
+	defer h.Cleanup()
+
+	h.TempDir("dep")
+
+	a := Analyzer{}
+
+	m, l, err := a.DeriveManifestAndLock(h.Path("dep"), "my/fake/project")
+	if m != nil || l != nil || err != nil {
+		t.Fatalf("expected manifest & lock & err to be nil: m -> %#v l -> %#v err-> %#v", m, l, err)
+	}
+}
+
+func TestAnalyzerDeriveManifestAndLockCannotOpen(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		// XXX: setting permissions works differently in
```

Comment:
Cool cool - yeah, I tend to use TODO for notes and FIXME for more glaring problems. TODO seems reasonable here. Thanks!